zenoh/src/net/routing/pubsub.rs

```diff
@@ -1567,7 +1577,11 @@ fn should_route(
     outface: &Arc<FaceState>,
     expr: &mut RoutingExpr,
 ) -> bool {
-    if src_face.id != outface.id {
+    if src_face.id != outface.id
+        && (src_face.mcast_group.is_none()
```

Comment:
The rhs of `&&`:

```rust
src_face.mcast_group.is_none() 
	|| outface.mcast_group.is_none() 
	|| src_face.mcast_group.as_ref().unwrap() != outface.mcast_group.as_ref().unwrap()
```

can be replaced with:

```rust
match (src_face.mcast_group.as_ref(), outface.mcast_group.as_ref()) {
    (None,_) | (_, None)=> true,
    (Some(l), Some(r)) => l != r,
}
```
This allows to remove unnecessary `unwrap()`.